packages/strapi-admin/admin/src/components/EventInput/index.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useGlobalContext } from 'strapi-helper-plugin';
+
+import Wrapper from './Wrapper';
+import EventRow from './EventRow';
+
+const EventInput = ({ onChange, name: inputName, value: inputValue }) => {
+  const { formatMessage } = useGlobalContext();
+
```

Comment:
If think it might be great to format your value so you get something similar to your events object.
This way you could simplify you `handleChangeAll` function also the `EventRow` component;